src/contentChunk.js

```diff
@@ -168,6 +170,36 @@ export const retrievePayload = async fileId => {
   })
 }
 
+export const retrieveBody = async (bodyId, range) => {
+  if (!bodyId) {
+    throw new Error('bodyID not supplied')
+  }
+
+  const fileDetails = await getFileDetails(bodyId)
+
+  if (!fileDetails) {
+    throw new Error('Could not file specified file')
+  }
+  if (range.start && range.start >= fileDetails.length) {
+    throw new Error('Start range cannot be greater than file length')
+  }
+  if (range.end && range.end > fileDetails.length) {
+    range.end = fileDetails.length
+  }
+
+  const contentEncoding = fileDetails ? (fileDetails.metadata ? fileDetails.metadata['content-encoding'] : null) : null
+  const decompressionStream = getDecompressionStreamByContentEncoding(contentEncoding)
+
+  const bucket = getGridFSBucket()
+  const downloadStream = bucket.openDownloadStream(bodyId, range)
+  downloadStream.on('error', err => {
+    logger.error(err)
```

Comment:
It might be better to throw error here instead of just logging out the error. Then we can inform the user of the error when trying to retrieve the body

Reply:
I agree but the issues is because we are using a stream and the error get's thrown async this will be thrown after the response has been sent and the body is streaming. I don't think there is a way to catch it and even if we do catch it, what do we do with it. So logging seemed like the only reasonable action to take. Unless you have a better idea.

Reply:
Ok cool, yeah thats a good point. Im not sure if this is possible but could we not add the error message as the last chuck in the stream sent to the client and then close the stream, Or just kill the stream which will fail on the client, and make the client request it again, hopefully with no errors. Not sure if this is possible or feasible though. Just thinking in terms of notifying the client that something happened instead of only getting half the requested payload back. But maybe this can be implemented another time

Reply:
I don't think we should pollute the stream with random error text. It might break the response even more. Especially if we just append text to what the client thinks is an image for example. I couldn't think of a good way to notify the user, so I think this is the best we can do.

Reply:
Ok cool that makes sense, Lets leave this as is for now then and can re-look into this more if this is picked up as an issue for clients. For now, The console will be the only real client making use of this endpoint so should be fine for the time being